src/common.rs

```diff
@@ -38,16 +38,36 @@ impl ColorType {
         }
     }
 
-    pub(crate) fn raw_row_length_from_width(&self, depth: BitDepth, width: u32) -> usize {
-        let bits = width as usize * self.samples() * depth as usize;
+    pub(crate) fn checked_raw_row_length(&self, depth: BitDepth, width: u32) -> Option<usize> {
```

Comment:
This whole function could just be:
```rust
   TryFrom::try_from((width as u64 * (self.samples() * depth as usize) as u64 + 7) / 8 + 1).ok()
```
LLVM is even able to produce [quite reasonable code](https://rust.godbolt.org/z/bWrmjd) for the on 32-bit targets, and there aren't issues with a hypothetical 128-bit target either because _width_ is a u32 so the full product can't overflow.

Reply:
Good point, introducing internal converters to `u8` (we can discuss making them public later) makes this even clearer. Then this multiplication is straightup `u32*u8*u8` which surely fits in `u64`.